modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/navigate.es.js

```diff
@@ -23,17 +23,23 @@
  */
 
 export default function (url, listeners) {
-	if (Liferay.SPA && Liferay.SPA.app && Liferay.SPA.app.canNavigate(url)) {
-		Liferay.SPA.app.navigate(url);
+	let urlString = url;
+
+	if (typeof url !== 'string') {
```

Comment:
I think We could migrate these utils to TS one day. We don't have any hint to say to users: "Hey, you could use a string or an URL there!". I think We may fighting this issue for non expected usages of navigate due to this :/

Reply:
Yeah, you're right. This is the perfect place that demonstrates value of TS, as bug happened exactly because of the unexpected argument type. We expected string, were not aware that `URL` object worked.

Also, after thinking about this a bit, it's not good to have overly permissive API. I'll rework type coercion to just allowing string and `URL`. Note - just allowing string is not a good option now, as we would also be needlessly running into backwards compatibility issue.